GH#37505: Adding control plane to ClusterAutoscaler resource definition (#38650)
openshift/openshift-docs: modules/cluster-autoscaler-about.adoc

The cluster autoscaler adjusts the size of an {product-title} cluster to meet its current deployment needs. It uses declarative, Kubernetes-style arguments to provide infrastructure management that does not rely on objects of a specific cloud provider. The cluster autoscaler has a cluster scope, and is not associated with a particular namespace.

The cluster autoscaler increases the size of the cluster when there are pods that failed to schedule on any of the current worker nodes due to insufficient resources or when another node is necessary to meet deployment needs. The cluster autoscaler does not increase the cluster resources beyond the limits that you specify.

> The cluster autoscaler computes the total memory, CPU, and GPU on all nodes the cluster, even tought it does not manage the control plane nodes. If you set the maximum memory resource limit, the autoscaler includes all the nodes in the cluster when calculating the current memory usage. It then uses that calculation to determine if the cluster autoscaler has the capacity to add more resources. These limits are not single-machine oriented. They are an aggregation of all the resources in the entire cluster. 

Reviewer: minor nit
```suggestion
The cluster autoscaler computes the total memory, CPU, and GPU on all nodes the cluster, even though it does not manage the control plane nodes. If you set the maximum memory resource limit, the autoscaler includes all the nodes in the cluster when calculating the current memory usage. It then uses that calculation to determine if the cluster autoscaler has the capacity to add more resources. These limits are not single-machine oriented. They are an aggregation of all the resources in the entire cluster. 
```

> The cluster autoscaler computes the total memory, CPU, and GPU on all nodes the cluster, even though it does not manage the control plane nodes. These values are not single-machine oriented. They are an aggregation of all the resources in the entire cluster. For example, if you set the maximum memory resource limit, the autoscaler includes all the nodes in the cluster when calculating the current memory usage. It then uses that calculation to determine if the cluster autoscaler has the capacity to add more worker resources.

Reviewer 2: ```suggestion
The cluster autoscaler computes the total memory, CPU, and GPU on all nodes the cluster, even though it does not manage the control plane nodes. These values are not single-machine oriented. They are an aggregation of all the resources in the entire cluster. For example, if you set the maximum memory resource limit, the autoscaler includes all the nodes in the cluster when calculating the current memory usage. That calculation is used to determine if the cluster autoscaler has the capacity to add more worker resources.
```

Trying to avoid using "human-like" qualities to a sofware/machine :)

Reviewer: > Trying to avoid using "human-like" qualities to a sofware/machine :)

just curious, to better help my learning, which part of that was reading as a "human-like" quality on the software/machine? (trying to avoid doing this in the future)

Author: @sagidlow I assumed you were referring to _ It then uses_ as human-like.

Reviewer: ahh, ok. that makes sense. thanks =)

* Pods that cannot be moved elsewhere because of a lack of resources, incompatible node selectors or affinity, matching anti-affinity, and so on.
* Unless they also have a `"cluster-autoscaler.kubernetes.io/safe-to-evict": "true"` annotation, pods that have a `"cluster-autoscaler.kubernetes.io/safe-to-evict": "false"` annotation.

When the cluster autoscaler computes the total memory, CPU, and GPU in the cluster, it does not differentiate between the worker or control plane nodes. If you set the maximum memory resource limit, the autoscaler includes all the nodes in the cluster when calculating the current memory usage. It then uses that calculation to determine if the cluster autoscaler has the capacity to add more resources. These limits are not single-machine oriented. They are an aggregation of all the resources in the entire cluster. 

> For example, you set the maximum CPU limit to 64 cores and set to only create machines that have 8 cores each. If your cluster starts has 30 cores, the cluster autoscaler can add up to 4 more nodes (with 32 cores).

Reviewer: i think a word or two got dropped
```suggestion
For example, you set the maximum CPU limit to 64 cores and set the cluster autoscaler to only create machines that have 8 cores each. If your cluster starts has 30 cores, the cluster autoscaler can add up to 4 more nodes (with 32 cores).
```

* Pods that cannot be moved elsewhere because of a lack of resources, incompatible node selectors or affinity, matching anti-affinity, and so on.
* Unless they also have a `"cluster-autoscaler.kubernetes.io/safe-to-evict": "true"` annotation, pods that have a `"cluster-autoscaler.kubernetes.io/safe-to-evict": "false"` annotation.

> For example, you set the maximum CPU limit to 64 cores and set the cluster autoscaler to only create machines that have 8 cores each. If your cluster starts has 30 cores, the cluster autoscaler can add up to 4 more nodes (with 32 cores).

Reviewer: minor nit
```suggestion
For example, you set the maximum CPU limit to 64 cores and set the cluster autoscaler to only create machines that have 8 cores each. If your cluster starts with 30 cores, the cluster autoscaler can add up to 4 more nodes (with 32 cores).
```
